google/cloud/spanner/samples/client_samples.cc

```diff
@@ -0,0 +1,212 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/spanner/admin/database_admin_client.h"
+#include "google/cloud/spanner/admin/instance_admin_client.h"
+#include "google/cloud/spanner/client.h"
+#include "google/cloud/spanner/testing/random_database_name.h"
+#include "google/cloud/spanner/testing/random_instance_name.h"
+#include "google/cloud/credentials.h"
+#include "google/cloud/internal/getenv.h"
+#include "google/cloud/internal/random.h"
+#include "google/cloud/testing_util/example_driver.h"
+#include <fstream>
+#include <string>
+#include <vector>
+
+namespace {
+
+void ClientSetEndpoint(std::vector<std::string> const& argv) {
+  if (argv.size() != 3) {
+    throw google::cloud::testing_util::Usage{
+        "client-set-endpoint <project-id> <instance-id> <database-id>"};
+  }
+  //! [client-set-endpoint]
+  namespace spanner = ::google::cloud::spanner;
+  [](std::string const& project_id, std::string const& instance_id,
+     std::string const& database_id) {
+    auto options = google::cloud::Options{}.set<google::cloud::EndpointOption>(
+        "private.googleapis.com");
+    return spanner::Client(spanner::MakeConnection(
+        spanner::Database(project_id, instance_id, database_id), options));
+  }
+  //! [client-set-endpoint]
+  (argv.at(0), argv.at(1), argv.at(2));
+}
+
+void ClientWithServiceAccount(std::vector<std::string> const& argv) {
+  if (argv.size() != 4) {
+    throw google::cloud::testing_util::Usage{
+        "client-with-service-account <project-id> <instance-id> <database-id> "
+        "<keyfile>"};
+  }
+  //! [client-with-service-account]
+  namespace spanner = ::google::cloud::spanner;
+  [](std::string const& project_id, std::string const& instance_id,
+     std::string const& database_id, std::string const& keyfile) {
+    auto is = std::ifstream(keyfile);
+    is.exceptions(std::ios::badbit);
+    auto contents = std::string(std::istreambuf_iterator<char>(is.rdbuf()), {});
+    auto options =
+        google::cloud::Options{}.set<google::cloud::UnifiedCredentialsOption>(
+            google::cloud::MakeServiceAccountCredentials(contents));
+    return spanner::Client(spanner::MakeConnection(
+        spanner::Database(project_id, instance_id, database_id), options));
+  }
+  //! [client-with-service-account]
+  (argv.at(0), argv.at(1), argv.at(2), argv.at(3));
+}
+
+void DatabaseAdminClientSetEndpoint(std::vector<std::string> const& argv) {
```

Comment:
same comment about: https://github.com/googleapis/google-cloud-cpp/blob/7ddabb92cb6619161d4d0c87b3962b9bde2b2670/google/cloud/spanner/admin/samples/database_admin_client_samples.cc#L32-L44

Reply:
Fixed.

Reply:
I would have expected these `InstanceAdminClient` and `DatabaseAdminClient` samples to have disappeared then.

Reply:
Ooops, fixed now.